nilearn/decoding/fista.py

```diff
@@ -184,7 +187,9 @@ def mfista(f1_grad, f2_prox, total_energy, lipschitz_constant, w_size,
 
         # backward (prox) step
         for _ in range(10):
-            w, prox_info = f2_prox(z - stepsize * gradient_buffer, stepsize,
+            # z <- -stepsize * gradient_buffer + z
+            z = inplace_add_mult(gradient_buffer, z, a=-stepsize)
```

Comment:
> makes sense - but we have full control over z as a local variable. Why
> would we let that happen? Or alternatively, shouldn't we warn about
> performance loss if this situation occurs?

I am just being extra careful. I'd rather take a small performance hit
rather than having code that might not work on a user's computer (eg
because that user uses a BLAS that I don't know).